src/framework/dockwindow/qml/Muse/Dock/DockTitleBar.qml

```diff
@@ -33,75 +33,112 @@ Item {
 
     required property QtObject titleBarCpp
 
-    property alias contextMenuModel: contextMenuButton.menuModel
-    property alias heightWhenVisible: titleBar.heightWhenVisible
+    property Component titleBarItem: defaultTitleBarComponent
+    property var contextMenuModel: null
+    property var heightWhenVisible: null
+    property var navigationPanel: null
+    property var navigationOrder: null
     property bool isHorizontalPanel: false
 
-    property alias navigation: contextMenuButton.navigation
-
     signal handleContextMenuItemRequested(string itemId)
 
     width: parent.width
     height: visible ? heightWhenVisible : 0
 
     visible: Boolean(titleBarCpp)
 
-    KDDW.TitleBarBase {
-        id: titleBar
-
+    Loader {
+        id: titleBarLoader
         anchors.fill: parent
 
-        heightWhenVisible: titleBarContent.implicitHeight
-        color: ui.theme.backgroundPrimaryColor
+        property var titleBarCpp: root.titleBarCpp
 
-        visible: parent.visible
+        sourceComponent: root.titleBarItem
 
-        MouseArea {
-            id: mouseArea
-            anchors.fill: parent
-            acceptedButtons: Qt.NoButton
-            cursorShape: Qt.SizeAllCursor
+        onLoaded: {
+            if (titleBarLoader.item) {
+                if (root.navigationPanel) {
+                    item.navigationPanel = root.navigationPanel
```

Comment:
I think should be `Qt.binding` for all the properties to avoid losing further property changes